extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java

```diff
@@ -104,36 +104,70 @@ public Result<Map<String, Object>> getVerifiedCredentials(DidDocument didDocumen
 
         var verifiedCredentials = verifyCredentials(verifiableCredentials, didDocument);
 
-        monitor.debug(() -> String.format("Verified %s credentials", verifiedCredentials.size()));
+        monitor.debug(() -> String.format("Verified %s credentials", verifiedCredentials.getContent().size()));
 
-        var claims = extractClaimsFromCredential(verifiedCredentials);
+        var claims = extractClaimsFromCredential(verifiedCredentials.getContent());
 
-        if (claims.failed()) {
+        var failureMessages = Stream.concat(verifiedCredentials.getFailureMessages().stream(),
+                claims.getFailureMessages().stream()).collect(Collectors.toList());
+
+        var result = new VerificationResult<>(claims.getContent(), failureMessages);
+
+        if (result.failed()) {
             monitor.severe(() -> String.format("Credentials verification failed: %s", claims.getFailureDetail()));
-            return Result.failure(claims.getFailureDetail());
+            return Result.failure(result.getFailureDetail());
         } else {
-            return Result.success(claims.getContent());
+            return Result.success(result.getContent());
         }
     }
 
     @NotNull
-    private List<SignedJWT> verifyCredentials(StatusResult<Collection<SignedJWT>> jwts, DidDocument didDocument) {
-        var result = jwts.getContent()
+    private VerificationResult<List<SignedJWT>> verifyCredentials(StatusResult<Collection<SignedJWT>> jwts, DidDocument didDocument) {
+        // Get valid credentials.
+        var verifiedJwts = jwts.getContent()
+                .stream()
+                .map(jwt -> verifyJwtClaims(jwt, didDocument))
+                .collect(partitioningBy(AbstractResult::succeeded));
+
+        var jwtsSignedByIssuer = verifiedJwts.get(true)
```

Comment:
For consistency with the block above:
```suggestion
        var jwtsSignedByIssuer = verifiedJwts.get(true)
                .stream()
                .map(jwt -> jwtCredentialsVerifier.isSignedByIssuer(jwt.getContent()))
                .collect(partitioningBy(AbstractResult::succeeded));
```